functions/tests/firestore-rules.test.ts

```diff
@@ -0,0 +1,289 @@
+import * as firebase from '@firebase/testing';
+import * as fs from 'fs';
+import { QuestionnaireType } from '../src/models/questionnaires';
+
+const projectId = 'reach-4-help-test';
+
+const rules = fs.readFileSync(`${__dirname}/../../firebase/firestore.rules`, 'utf8');
+
+/**
+ * Creates a new app with authentication data matching the input.
+ *
+ * @param {object} auth the object to use for authentication (typically {uid: some-uid})
+ * @return {object} the app.
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const authedApp = (auth: any) => {
+  return firebase
+    .initializeTestApp({ projectId, auth })
+    .firestore();
+};
+
+beforeAll(async () => {
+  await firebase.loadFirestoreRules({ projectId, rules });
+});
+
+beforeEach(async () => {
+  // Clear the database between tests
+  await firebase.clearFirestoreData({ projectId });
+});
+
+afterAll(async () => {
+  await Promise.all(firebase.apps().map(app => app.delete()));
+});
+
+describe('users', () => {
+  it('require users to log in before reading a profile', async () => {
+    const db = authedApp(null);
+    const profile = db.collection('users').doc('1234');
+    await firebase.assertFails(profile.get());
+  });
+
+  it('require users to log in before listing profiles', async () => {
+    const db = authedApp(null);
+    const profile = db.collection('users');
+    await firebase.assertFails(profile.get());
+  });
+
+  it('authenticated users can read other users profiles', async () => {
+    const db = authedApp({ uid: '1234' });
+    const profile = db.collection('users').doc('5678');
+    await firebase.assertSucceeds(profile.get());
+  });
+
+  it('authenticated users can list users profiles', async () => {
```

Comment:
Same as above, we should do the opposite test with an unauthenticated user.

Reply:
```typescript
  it('require users to log in before listing profiles', async () => {
    const db = authedApp(null);
    const profile = db.collection('users');
    await firebase.assertFails(profile.get());
  });
```